pkg/llmstrategy/interfaces.go

```diff
@@ -16,10 +16,22 @@ package llmstrategy
 
 import (
 	"context"
+	"io"
 
 	"github.com/GoogleCloudPlatform/kubectl-ai/pkg/ui"
 )
 
 type Strategy interface {
-	RunOnce(ctx context.Context, query string, previousQueries []string, userInterface ui.UI) error
+	// NewConversation starts a new conversation with the LLM.
+	// A conversation has state that persists across rounds.
+	NewConversation(ctx context.Context, userInterface ui.UI) (Conversation, error)
+}
+
+type Conversation interface {
+	// Close should be called to free up resources
+	io.Closer
+
+	// RunOneRound will send the query to the LLM, and go through cycles with the LLM,
+	// evaluating requested functions until we reach a stopping point.
+	RunOneRound(ctx context.Context, query string) error
```

Comment:
Something doesn't feel right about these interfaces.

So my mental model is:

I want to have a conversation with a particular style (ReAct or ChatBased), so one way to represent will be:

```

conversationStyle = ReAct{} or ChatBased{}
_ = conversation.Do(style ReActOrChatBased)

```